src/main/java/engineering/swat/watch/impl/JDKPoller.java

```diff
@@ -83,32 +86,38 @@ private static void poll() {
     }
 
 
-    public static Closeable register(Path path, Consumer<List<WatchEvent<?>>> changes) throws IOException {
+    public static Closeable register(SubscriptionKey path, Consumer<List<WatchEvent<?>>> changesHandler) throws IOException {
         logger.debug("Register watch for: {}", path);
 
         try {
             return CompletableFuture.supplyAsync(() -> {
-                    try {
-                        return path.register(service, ENTRY_CREATE, ENTRY_MODIFY, ENTRY_MODIFY, ENTRY_DELETE);
-                    } catch (IOException e) {
-                        throw new RuntimeException(e);
+                try {
+                    WatchEvent.Kind<?>[] kinds = new WatchEvent.Kind[]{ ENTRY_CREATE, ENTRY_MODIFY, ENTRY_MODIFY, ENTRY_DELETE };
```

Comment:
ah, good catch. no idea if it can be final, especially since array contents of final arrays can still be modified.

Reply:
alternativly we would offer the user a way to only request certain event types, but that changes quite a bit in the internals. so that looks more like a future addition.